docs/api.md

```diff
@@ -2089,6 +2090,17 @@ const [response] = await Promise.all([
 
 Shortcut for [page.mainFrame().waitForNavigation(options)](#framewaitfornavigationoptions).
 
+#### page.waitForNetworkIdle([options])
+- `options` <[Object]> Optional waiting parameters
+  - `timeout` <[number]> Maximum wait time in milliseconds, defaults to 30 seconds, pass `0` to disable the timeout. The default value can be changed by using the [page.setDefaultTimeout(timeout)](#pagesetdefaulttimeouttimeout) method.
+  - `idleTime` <[number]> How long to wait for no network requests in milliseconds, defaults to 500 milliseconds.
+- returns: <[Promise]<void>> Promise which resolves when network is idle.
+
+```js
+page.evaluate(() => fetch('some-url'));
```

Comment:
Why do we need to use `fetch` here? 
Can't this work with `page.goto` like below?

```
[response] = await Promise.all([
            await page.goto(url, {'timeout': timeOut, 'waitUntil': 'load']}),
            await page.waitForNetworkIdle({idleTime:1000, timeout:30000})
 ]);
```

I did some testing with some URLs and it's not working as expected.

Reply:
I was trying to come up with a simple example. One case though is you update the css for something that results in downloading an image, and then you want to wait for the image to be downloaded before you continue and take a screenshot 

What wasn’t working as expected?